index.ts

```diff
@@ -3,6 +3,7 @@ const NativeClient = require('bindings')('hawk_tracer_client').HawkTracerClient;
 export interface HawkTracerClientOptions {
     source: string;
     map_files?: string;
+    wait_for_server?: boolean;
```

Comment:
We don't need this now.

Reply:
I think it's important for us to provide a maximum retries configuration here instead, it can be optional, but that way we can ensure that connection attempt doesn't run forever if there is no need for it

Reply:
Sure, does a default of 100 retries sound reasonable?

Reply:
Also @beila do you agree?

Reply:
Okay, thinking more about it, it's a actually a bit tricky – because the retries are done on the JS side (it's simpler), there is no point passing the `max_retries` to the CPP code, to be honest (at least in the current shape).

Reply:
I agree to @Artie18. If CPP doesn't need it, you don't have to pass it.

Reply:
This should be done now, with a `max_retries` argument (default to 100)